File: 006/006.cc
#include <iostream>
#include <cmath>

using namespace std;

/* Quickly sums numbers form 1 to N */
int quicksum(int n)
{
    return (n + 1) * n / 2;
}

int sumOfSquares(int n)
{
    int sum = 0;
    for(int i=1; i <= n; ++i) {
        sum += i*i;
    }

    return sum;
}

int diff(int n)
{
    return pow(quicksum(n), 2) - sumOfSquares(n);
}

int main()
{
    cout << "Diff(10): " << diff(10) << endl;
    cout << "Diff(100): " << diff(100) << endl;
    return 0;
}

File: 006/006.hs
module Main where

-- Quickly sums numbers form 1 to N
quicksum :: Integer -> Integer
quicksum n = (n + 1) * n `div` 2

diff :: Integer -> Integer
diff n = toInteger $ floor $ (fromInteger (quicksum n) :: Float) ** 2 - sumOfSquares n
    where
      sumOfSquares n = sum $ map (\x -> (fromInteger x) ** 2) [1..n]
             
main = do
  putStrLn $ "Diff(10): " ++ show (diff 10)
  putStrLn $ "Diff(10): " ++ show (diff 100)
